src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -590,22 +582,52 @@ public Object evalBindingCached(
                 @Cached("create(cachedCallTarget)") DirectCallNode callNode,
                 @Cached("create()") RopeNodes.EqualNode equalNode) {
             final MaterializedFrame parentFrame = BindingNodes.getTopFrame(binding);
+            return eval(cachedRootNode, cachedCallTarget, callNode, parentFrame);
+        }
+
+        private Object eval(final RootNodeWrapper rootNode, final CallTarget callTarget, final DirectCallNode callNode, final MaterializedFrame parentFrame) {
```

Comment:
Please remove the `final` in arguments, it's extra noise and bad practice to re-assign arguments anyway.